R/utils.R

```diff
@@ -132,8 +132,12 @@ eval_with_cols = function(orig_call, all_cols) {
   })
   if (!is.primitive(fun)) {
     named_call = match.call(fun, orig_call)
-    if ("cols" %in% names(formals(fun)) && !"cols" %in% names(named_call)) {
-      named_call[["cols"]] = all_cols
+    if ("cols" %in% names(formals(fun))) {
+      if ("cols" %in% names(named_call)) {
+        stopf("user should not provide cols argument to %s", paste(fun_uneval))
```

Comment:
Can `length(fun_uneval)>1`? It looks like it's coming from `substitute(call)[[1]]`. That can be `length>1` in namespaced calls:

```r
length(substitute(stats::lm(x))[[1]])
[1] 3
```

Is that possibly the input here? If that case was not considered previously we might file a follow-up bug to handle this.

Relevant to the current PR because that `paste()` call might not WAI.

In any case, I would prefer `as.character()` to make the intention explicit.

Reply:
Second observation: this error will not be very actionable. As a "clueless" user I may get this error and worry "What did I do wrong? What does this mean?" More helpful might be to suggest an alternative, or point to a helpful doc page.

Reply:
> Is that possibly the input here?

Answering my own question based on the tests: Looks like no. But we should add `stopifnot(length(fun_uneval) == 1)` -- most importantly we might eventually export placeholders for package authors (like we do with `` `:=` ``) as masks for `R CMD check` issues, and then authors might be tempted to use `data.table::measure` leading to the above issue.

Reply:
ok i added the stopifnot and I changed paste to as.character.